yaks/lib/yaks/mapper/attribute.rb

```diff
@@ -2,22 +2,23 @@ module Yaks
   class Mapper
     class Attribute
       extend Forwardable, Util
-      include Util
-      include Attribs.new(:name, :block, options: {}.freeze)
+      include Attribs.new(:name, :block, if: true), Util
 
-      def self.create(*args, &block)
-        args, options = extract_options(args)
-        new(name: args.first, options: options, block: block)
+      def self.create(name, options = {}, &block)
+        options[:name] = name
+        options[:block] = block
+        new(options)
       end
 
       def add_to_resource(resource, mapper, _context)
-        return unless mapper.expand_value(options.fetch(:if, true))
-
+        return resource unless Resolve(self.if, mapper)
```

Comment:
I think there should be another spec to test the proc/lambda is actually `instance_eval/exec` on the mapper.